### PR TITLE
feat(s3): add s3_bucket_object_lock check

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add `s3_bucket_object_lock` check to verify if a S3 bucket has configured WORM (Write Once Read Many). That helps to prevent ransomware attacks.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock.metadata.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "s3_bucket_object_lock",
+  "CheckTitle": "Check if S3 buckets have object lock enabled",
+  "CheckType": [
+    "Data Protection"
+  ],
+  "ServiceName": "s3",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock.metadata.json]: Could you please fill this with the S3 bucket ARN template?

File: tests/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock_test.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                # Test Check
+                from prowler.providers.aws.services.s3.s3_bucket_object_lock.s3_bucket_object_lock import (
+                    s3_bucket_object_lock,
+                )
+
+                check = s3_bucket_object_lock()
+                result = check.execute()
+
+                assert len(result) == 1
+                assert result[0].status == "FAIL"
```
Comment (jfagoagas) [inline tests/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock_test.py:89]: Please assert also the tags here.

File: tests/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock_test.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                from prowler.providers.aws.services.s3.s3_bucket_object_lock.s3_bucket_object_lock import (
+                    s3_bucket_object_lock,
+                )
+
+                check = s3_bucket_object_lock()
+                result = check.execute()
+
+                assert len(result) == 1
+                assert result[0].status == "PASS"
+                assert search(
```
Comment (jfagoagas) [inline tests/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock_test.py:134]: Please assert also the tags here.

File: tests/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock_test.py
Comment (jfagoagas) [inline tests/providers/aws/services/s3/s3_bucket_object_lock/s3_bucket_object_lock_test.py:1]: Could you include the test case for zero buckets and zero findings?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Awesome!

Check my comments to merge this. Thanks!

**Review by jfagoagas — DISMISSED** [review]
🚀

**Review by jfagoagas — APPROVED** [review]
